op-batcher/batcher/driver.go

```diff
@@ -550,9 +562,40 @@ func (l *BatchSubmitter) cancelBlockingTx(queue *txmgr.Queue[txRef], receiptsCh
 
 // sendTransaction creates & queues for sending a transaction to the batch inbox address with the given `txData`.
 // The method will block if the queue's MaxPendingTransactions is exceeded.
-func (l *BatchSubmitter) sendTransaction(ctx context.Context, txdata txData, queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef]) error {
+func (l *BatchSubmitter) sendTransaction(ctx context.Context, txdata txData, queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef], daWaitGroup *sync.WaitGroup, daSemaphore *semaphore.Weighted) error {
 	var err error
-	// Do the gas estimation offline. A value of 0 will cause the [txmgr] to estimate the gas limit.
+
+	// if Alt DA is enabled we post the txdata to the DA Provider and replace it with the commitment.
+	if l.Config.UseAltDA {
+		// sanity checks
+		if nf := len(txdata.frames); nf != 1 {
+			l.Log.Crit("Unexpected number of frames in calldata tx", "num_frames", nf)
+		}
+		if txdata.asBlob {
+			l.Log.Crit("Unexpected blob txdata with AltDA enabled")
+		}
+
+		// when posting txdata to an external DA Provider, we use a goroutine to avoid blocking the main loop
+		// since it may take a while for the request to return.
+		daWaitGroup.Add(1)
+		go func() {
+			defer daWaitGroup.Done()
+			defer daSemaphore.Release(1)
```

Comment:
defer is last-in first-out. Should the order be switched? first say it is done, then release

Reply:
https://cs.opensource.google/go/x/sync/+/refs/tags/v0.8.0:errgroup/errgroup.go;l=38
Seems like errgroup also releases the sem before noticing the waitgroup.
Not sure if either matters... but I guess we should do whatever errgroup is doing? I can add a comment

Reply:
actually switched to an errgroup.